package protoStream;

/*
 * protoc --proto_path=src/proto --java_out=src/java src/proto/protoStream.proto
 */

option java_package         = "com.spinn3r.api.protobuf";
option java_outer_classname = "ProtoStream";

message ApplicationHeader {
  required string name  = 1;
  optional string value = 2;
}

message ProtoStreamHeader {
  required string version            = 1; // version of the protostream format
  required string default_entry_type = 2; // the type of entries in this stream

  // application specific header fields
  repeated ApplicationHeader application_header = 3;
}

message ProtoStreamDelimiter {

  enum DelimiterType {
    ENTRY = 1;
    END   = 2;
    INDEX = 3;
  }

  optional DelimiterType delimiter_type = 1 [default = END];
}
